api/core/repository/repository_factory.py

```diff
@@ -0,0 +1,97 @@
+"""
+Repository factory for creating repository instances.
+
+This module provides a simple factory interface for creating repository instances.
+It does not contain any implementation details or dependencies on specific repositories.
+"""
+
+from collections.abc import Callable, Mapping
+from typing import Any, Literal, Optional, cast
+
+from core.repository.workflow_node_execution_repository import WorkflowNodeExecutionRepository
+
+# Type for factory functions - takes a dict of parameters and returns any repository type
+RepositoryFactoryFunc = Callable[[Mapping[str, Any]], Any]
+
+# Type for workflow node execution factory function
+WorkflowNodeExecutionFactoryFunc = Callable[[Mapping[str, Any]], WorkflowNodeExecutionRepository]
+
+# Repository type literals
+RepositoryType = Literal["workflow_node_execution"]
```

Comment:
Since the `_registry_factory` and `_create_factory` methods in the `RepositoryFactory` are intended for internal use and marked as private, I suggest converting the `RepositoryType` into a private member as well to maintain consistency and encapsulation within the module.